vyper/ast/folding.py

```diff
@@ -172,20 +174,40 @@ def replace_user_defined_constants(vyper_module: vy_ast.Module) -> int:
             # annotation is not wrapped in `constant(...)`
             continue
 
-        changed_nodes += replace_constant(vyper_module, node.target.id, node.value, False)
+        # Extract typedef for propagation to type checker
+        node_annotation_arg = node.get("annotation.args")[0]
+        type_: Union[BaseTypeDefinition, None] = None
+        try:
+            type_ = get_type_from_annotation(node_annotation_arg, DataLocation.STORAGE)
+        except:  # noqa: E722
```

Comment:
Can we except `VyperException`?
https://github.com/vyperlang/vyper/blob/master/vyper/exceptions.py#L29